pyresample/bilinear/__init__.py

```diff
@@ -210,97 +203,448 @@ def get_bil_info(source_geo_def, target_area_def, radius=50e3, neighbours=32,
+        """Set up kd tree on input."""
+        valid_input_index, input_coords = self._get_valid_input_index_and_input_coords()
+        kdtree = None
+        if input_coords.size:
+            if nprocs > 1:
+                kdtree = kdtree_class(input_coords, nprocs=nprocs)
+            else:
+                kdtree = KDTree(input_coords)
+        return valid_input_index, kdtree
+
+    def _get_valid_input_index_and_input_coords(self):
+        valid_input_index, source_lons, source_lats = \
+            _get_valid_input_index(self._source_geo_def,
+                                   self._target_geo_def,
+                                   self._reduce_data,
+                                   self._radius_of_influence)
+        input_coords = lonlat2xyz(source_lons, source_lats)
+        valid_input_index = np.ravel(valid_input_index)
+        input_coords = input_coords[valid_input_index, :].astype(np.float)
+
+        return valid_input_index, input_coords
+
+    def get_sample_from_bil_info(self, data, fill_value=None, output_shape=None):
+        """Resample using pre-computed resampling LUTs."""
+        del output_shape
+        fill_value = _check_fill_value(fill_value, data.dtype)
+
+        res = _resample(
+            self._slice_data(data, fill_value),
+            (self.bilinear_s, self.bilinear_t)
+        )
+
+        return self._finalize_output_data(data, res, fill_value)
+
+    def _slice_data(self, data, fill_value):
+        return get_slicer(data)(data, self.slices_x, self.slices_y, self.mask_slices, fill_value)
+
+    def _finalize_output_data(self, data, res, fill_value):
+        return res
+
+    def save_bil_info(self, filename):
+        """Save bilinear resampling look-up tables."""
+        raise NotImplementedError
+
+    def load_bil_info(self, filename):
+        """Load bilinear resampling look-up tables and initialize the resampler."""
+        raise NotImplementedError
+
+
+def _check_fill_value(fill_value, dtype):
+    """Check that fill value is usable for the data."""
+    if fill_value is None:
+        if np.issubdtype(dtype, np.integer):
+            fill_value = 0
+        else:
+            fill_value = np.nan
+    elif np.issubdtype(dtype, np.integer):
+        if np.isnan(fill_value):
+            fill_value = 0
+        elif np.issubdtype(type(fill_value), np.floating):
+            fill_value = int(fill_value)
+
+    return fill_value
+
+
+def _get_output_xy(target_geo_def):
+    out_x, out_y = target_geo_def.get_proj_coords()
+    return np.ravel(out_x),  np.ravel(out_y)
+
+
+def _get_input_xy(source_geo_def, proj, valid_input_index, index_array):
+    """Get x/y coordinates for the input area and reduce the data."""
+    return proj(
+        *array_slice_for_multiple_arrays(
+            index_array,
+            array_slice_for_multiple_arrays(
+                valid_input_index,
+                mask_coordinates(*_get_raveled_lonlats(source_geo_def))
```

Comment:
I think you could use a couple of explanatory variables here to make it more readable.

Reply:
Yeah, certainly. I just blindly followed the refactoring book you introduced us to :wink:

Reply:
Don't hesitate to read it until the end, it a great book :) But refactoring.guru sums it up pretty well! And there you have a list of all the smells and refactorings

Reply:
Only have the free opening chapter, but I'll check the web page when I have the time. I'll unwrap the above (and similar) places a bit.

Reply:
Explanatory intermediate variables added to few functions/methods that got too far the road.